### ISSUE TITLE
Umlauts like Ä, Ö, Ü in Mailadresses/URLs are not detected as normal letters

### ISSUE THREAD

**Issue by leanderbuerkin** [DavidAnson/markdownlint#1212]
The Mailaddress
```
<küche@tf.de>
```
is automatically formatted to
```
<kü<che@tf.de>>
```.

Is that intended?

**Comment by DavidAnson** [comment]
MD034 reports: Bare URL used [Context: "che@tf.de"]

So it looks like the umlaut character is not allowed in an email address. That appears to be the case for the parser which limits that section to ASCII alphanumeric: https://github.com/micromark/micromark-extension-gfm-autolink-literal?tab=readme-ov-file#syntax

Rendering of the sample input via the parser markdownlint uses is the same as what happens on GitHub - the first two characters are not part of the link: https://dlaa.me/markdownlint/#%25m%3Ck%C3%BCche%40tf.de%3E%0A

So I think the current behavior is correct (and consistent with GitHub) based on how email addresses are defined and corresponding autolink behavior.

**Comment by leanderbuerkin** [comment]
Alright, we could solve our problem from the other direction: <kueche@tf.de> is forwarded to the same e-mailaddress (or is the same address?).

**Comment by DavidAnson** [comment]
I doubt that would go to the same person. I think the recommendation is probably to "percent encode" the umlaut character. or maybe ask the person who owns that alias as this is probably something they have dealt with before.

https://en.m.wikipedia.org/wiki/Percent-encoding

**Comment by DavidAnson** [comment]
Great, thank you for the update!
